### DISCUSSION TITLE
Sunbird User Org Service : User is created but password couldn't be updated.

### DISCUSSION THREAD

**Discussion by pritirc** [Sunbird-Lern/Community#64] [Issues] [answered]
Hi Team,

We have installed sunbird user org service from sunbird LMS module and installed required S/W KeyCloak SPI , Elastic Search & Cassandra DB .

When we are trying to register  user by using **/v1/user/signup** API , we found the response as follows .
{
    "id": "api.user.signup",
    "ver": "v1",
    "ts": "2023-01-23 17:57:25:552+0530",
    "params": {
        "resmsgid": "42bf41bd-6daa-4290-8481-1f5873a7fb6d",
        "msgid": "42bf41bd-6daa-4290-8481-1f5873a7fb6d",
        "err": null,
        "status": "SUCCESS",
        "errmsg": null
    },
    "responseCode": "OK",
    "result": {
        "response": "SUCCESS",
        "err_msg": "User is created but password couldn't be updated.",
        "userId": "2b457634-6853-4594-9d14-7fa58ba7b400"
    }
}

Could you help us to fix the above issues.

**Comment by reshmi-nair** [comment]
@pritirc Pls refer this thread to solve this issue.  https://github.com/orgs/Sunbird-Ed/discussions/306

**Reply by pritirc** [reply]
@reshmi-nair  When we try to create an user , There is no such method to add user in keycloak server. There will not be any user in keycloak then how update password will work. Is there something that I miss to add user into keycloak while registration happens.

**Reply by reshmi-nair** [reply, marked as answer]
@pritirc You need to configure the keycloak with Sunbird-Ed for user creation to happen in keycloak. User signup API only creates user in UserOrg DB and updates the password to keycloak user that is already created.
